gdal/scripts/fix_typos.sh

```diff
@@ -48,18 +48,18 @@ cd "$GDAL_ROOT"
 if ! test -d fix_typos; then
     # Get our fork of codespell that adds --words-white-list and full filename support for -S option
     mkdir fix_typos
-    cd fix_typos
-    git clone https://github.com/rouault/codespell
-    git checkout codespell/gdal_improvements
```

Comment:
This was actually broken from one of your previous changes.

```git checkout codespell/gdal_improvements``` errors out for me with
```error: pathspec 'codespell/gdal_improvements' did not match any file(s) known to git.```
The previous version that cd'ed to codespell and did git checkout gdal_improvements is the only one to work

Reply:
Fixed.